tests/integration/golang/mlflow/run/create_test.go

```diff
@@ -54,6 +54,94 @@ func (s *CreateRunTestSuite) Test_Ok() {
 	})
 	assert.Nil(s.T(), err)
 
+	s.testCases(namespace, experiment, false, *experiment.ID)
+}
+
+func (s *CreateRunTestSuite) Test_DefaultNamespaceExperimentZero_Ok() {
+	defer func() {
+		assert.Nil(s.T(), s.NamespaceFixtures.UnloadFixtures())
+	}()
+
+	// create test experiment.
+	namespace, err := s.NamespaceFixtures.CreateNamespace(context.Background(), &models.Namespace{
+		ID:                  1,
+		Code:                "default",
+		DefaultExperimentID: common.GetPointer(int32(0)),
+	})
+	assert.Nil(s.T(), err)
+
+	experiment, err := s.ExperimentFixtures.CreateExperiment(context.Background(), &models.Experiment{
+		Name:           uuid.New().String(),
+		NamespaceID:    namespace.ID,
+		LifecycleStage: models.LifecycleStageActive,
+	})
+	assert.Nil(s.T(), err)
+
+	// update default experiment id for namespace.
+	namespace.DefaultExperimentID = experiment.ID
+	_, err = s.NamespaceFixtures.UpdateNamespace(context.Background(), namespace)
+	assert.Nil(s.T(), err)
+
+	s.testCases(namespace, experiment, false, int32(0))
+}
+
+func (s *CreateRunTestSuite) Test_CustomNamespace_Ok() {
+	defer func() {
+		assert.Nil(s.T(), s.NamespaceFixtures.UnloadFixtures())
+	}()
+
+	// create test experiment.
+	namespace, err := s.NamespaceFixtures.CreateNamespace(context.Background(), &models.Namespace{
+		ID:                  1,
+		Code:                "custom",
+		DefaultExperimentID: common.GetPointer(int32(0)),
+	})
+	assert.Nil(s.T(), err)
+
+	experiment, err := s.ExperimentFixtures.CreateExperiment(context.Background(), &models.Experiment{
+		Name:           uuid.New().String(),
+		NamespaceID:    namespace.ID,
+		LifecycleStage: models.LifecycleStageActive,
+	})
+	assert.Nil(s.T(), err)
+
+	s.testCases(namespace, experiment, true, *experiment.ID)
+}
+
+func (s *CreateRunTestSuite) Test_CustomNamespaceExperimentZero_Ok() {
+	defer func() {
+		assert.Nil(s.T(), s.NamespaceFixtures.UnloadFixtures())
+	}()
+
+	// create test experiment.
+	namespace, err := s.NamespaceFixtures.CreateNamespace(context.Background(), &models.Namespace{
+		ID:                  1,
+		Code:                "custom",
+		DefaultExperimentID: common.GetPointer(int32(0)),
+	})
+	assert.Nil(s.T(), err)
+
+	experiment, err := s.ExperimentFixtures.CreateExperiment(context.Background(), &models.Experiment{
+		Name:           uuid.New().String(),
+		NamespaceID:    namespace.ID,
+		LifecycleStage: models.LifecycleStageActive,
+	})
+	assert.Nil(s.T(), err)
+
+	// update default experiment id.
+	namespace.DefaultExperimentID = experiment.ID
+	_, err = s.NamespaceFixtures.UpdateNamespace(context.Background(), namespace)
+	assert.Nil(s.T(), err)
+
+	s.testCases(namespace, experiment, true, int32(0))
+}
+
+func (s *CreateRunTestSuite) testCases(
```

Comment:
the naming is a bit strange of such function. is hard to say what is going on here and why do we need it.

Reply:
also I don't understand what this function tests? it just creates `Run` and that's it. We don't test `error` or `response` or that `run` has been actually created.

Reply:
Well the change in this file was to take the test logic from the old Test_OK and make it reusable with different namespace/experiment inputs. I didn't really change the interior of those tests.

Reply:
Just call it "testSuccessCases" and we can always improve it later.

Reply:
named it "successCases"

Reply:
Hooray